version: "3.3"

services:

  server:
    # image: data_model:latest
    image: njthykkathu/vis_test:latest
    command: redis-server /etc/redis/redis.conf --protected-mode no
    networks:
      - outside
    deploy:
      replicas: 1
      restart_policy: 
        condition: none
      placement:
        constraints:
          - node.labels.sip_worker6 == true
      labels:
        com.dn.description: "Host networking"

  config:
    # image: data_model:latest
    image: njthykkathu/vis_test:latest
    entrypoint: python3
    command: /home/sdp/redis/config_data.py
    networks:
      - outside
    depends_on:
      - server
    deploy:
      replicas: 1
      restart_policy: 
        condition: none
      placement:
        constraints:
          - node.labels.sip_worker6 == true
      labels:
        com.dn.description: "Host networking"

  test:
    # image: data_model:latest
    image: njthykkathu/vis_test:latest
    entrypoint: python3
    command: /home/sdp/redis/test_config_database.py
    networks:
      - outside
    depends_on:
      - server
    deploy:
      replicas: 1
      restart_policy: 
        condition: none
      placement:
        constraints:
          - node.labels.sip_worker6 == true
      labels:
        com.dn.description: "Host networking"

networks:
  outside:
    external:
      name: "host"

